migration/incremental/015-tracing-compression.sql

```diff
@@ -15,8 +15,16 @@ ANALYZE _ps_trace.instrumentation_lib;
 TRUNCATE TABLE _ps_trace.schema_url CASCADE;
 ANALYZE _ps_trace.schema_url;
 
-ALTER TABLE _ps_trace.tag DROP CONSTRAINT tag_key_value_id_key_id_key;
-CREATE UNIQUE INDEX tag_key_value_id_key_id_key ON _ps_trace.tag (key, _prom_ext.jsonb_digest(value)) INCLUDE (id, key_id);
+CREATE TABLE _ps_trace.tag (
+    id BIGINT NOT NULL GENERATED ALWAYS AS IDENTITY,
+    key_id BIGINT NOT NULL,
+    tag_type ps_trace.tag_type NOT NULL,
+    key ps_trace.tag_k NOT NULL REFERENCES _ps_trace.tag_key (key) ON DELETE CASCADE,
+    value ps_trace.tag_v NOT NULL
+);
+CREATE UNIQUE INDEX tag_key_value_id_key_id_key_idx ON _ps_trace.tag (key, _prom_ext.jsonb_digest(value)) INCLUDE (id, key_id);
```

Comment:
Did you happen to benchmark storing the digest in a column vs a functional index?

Reply:
No, I didn't.

Reply:
if we have to change the jsonb_digest function, will we have to drop and recreate the functional index?

I'm wondering whether the function is so fast that storing the digest in a column is not worth the storage overhead, or if storing it would make a noticeable difference in query performance. WDYT?

Reply:
> if we have to change the jsonb_digest function, will we have to drop and recreate the functional index?

Correct.

> I'm wondering whether the function is so fast that storing the digest in a column is not worth the storage overhead, or if storing it would make a noticeable difference in query performance. WDYT?

If we don't store it, how do we enforce the uniqueness constraint?

Reply:
it's stored in the index, but not the table. I was thinking about the difference between the two statements below, however I now realize that there's no way around doing the digest on the right side, and postgres just looks at the value stored in the index for the left side (it doesn't have to recompute). So, I withdraw my question. lol

`select * from tag t where jsonb_digest(t.value) = jsonb_digest(_thing_we_want_to_find);`

`select * from tag t where t.digest =  jsonb_digest(_thing_we_want_to_find);`